python/cudf/cudf/core/index.py

```diff
@@ -2158,6 +2164,27 @@ def __init__(
 
         super().__init__(data, **kwargs)
 
+        if self._freq is not None:
+            unique_vals = self.diff().unique()
+            if len(unique_vals) != 1 or unique_vals[0] != self._freq:
+                raise ValueError()
+
+    @_cudf_nvtx_annotate
+    def _copy_type_metadata(
+        self: DatetimeIndex, other: DatetimeIndex, *, override_dtypes=None
+    ) -> GenericIndex:
+        super()._copy_type_metadata(other, override_dtypes=override_dtypes)
+        self._freq = other._freq
```

Comment:
Unlike `_from_data` and `__init__`, it appears this one doesn't validate that the provided frequency is valid?